master/internal/experiment.go

```diff
@@ -252,20 +253,16 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 		resources.SetMaxSlots(msg.MaxSlots)
 		e.Config.SetResources(resources)
 		msg.Handler = ctx.Self()
-		ctx.Tell(e.rm, msg)
-	case sproto.SetGroupWeight:
-		resources := e.Config.Resources()
-		resources.SetWeight(msg.Weight)
-		e.Config.SetResources(resources)
-		msg.Handler = ctx.Self()
-		ctx.Tell(e.rm, msg)
-	case sproto.SetGroupPriority:
-		resources := e.Config.Resources()
-		resources.SetPriority(msg.Priority)
-		e.Config.SetResources(resources)
 		msg.Handler = ctx.Self()
 		ctx.Tell(e.rm, msg)
-
+	case job.SetGroupWeight:
+		if err := e.setWeight(ctx, msg.Weight); err != nil {
+			ctx.Log().WithError(err)
```

Comment:
we probably should change this to arrive here with an Ask ratherthan Tell so we can respond and return the error to the user.